gene2phenotype_project/gene2phenotype_app/serializers/publication.py

```diff
@@ -335,24 +335,19 @@ def create(self, validated_data):
         comment = None
         families = None
 
-        publication_data = validated_data.get('publication') # includes 'pmid', 'comment', 'families'
+        publication_data = self.initial_data.get('publication') # 'publication': {'pmid': 39385417}
+        comment = self.initial_data.get('comment', None) # 'comment': {'comment': 'this is a comment', 'is_public': 1}
+        families = self.initial_data.get("families", None) # "families": { "families": 200, "consanguinity": "unknown", "ancestries": "african", "affected_individuals": 2 }
 
-        if "comment" in publication_data:
-            comment = publication_data.get("comment")
+        if comment:
+            comment_text = comment.get("comment", None)
 
-            # Check if comment text is invalid or missing
-            if not comment or "comment" not in comment or comment.get("comment") == "":
+            # Check if comment text is empty string
+            if not comment_text or comment_text == "":
```

Comment:
Suggesting that this is just 
`if not comment_text:`

Reply:
In some texts, I inserted empty strings which made the code fail. To be sure that doesn't happen I added the check here.